src/containers/Main/Tabs/Screens/Home/connect.ts

```diff
@@ -1,19 +1,45 @@
-import { useCallback } from 'react';
+import { useCallback, useMemo, useState } from 'react';
 import { useNavigation } from '@react-navigation/core';
+import { useLessonsQuery } from 'apollo/generated/lessons';
+import { normalizeLessons } from 'models/Lesson';
 
 const useConnect = () => {
-  const { navigate, goBack, canGoBack } = useNavigation();
+  const { goBack, canGoBack } = useNavigation();
+
+  const { data } = useLessonsQuery();
+
+  const { lessons } = normalizeLessons(data);
```

Comment:
This normalization must occur directly on apollo hook. Remove this and make the proper normalizations directly on `useLessonsQuery` hook

Reply:
I have created a hook in the new commit where I do this but I am not sure about if it is done properly

Reply:
@josarcsal it's correct, don't worry